src/qilib/utils/storage/interface.py

```diff
@@ -109,6 +109,13 @@ def _tag_to_string(tag: Union[str, List[str]]) -> str:
             tag = '/'.join(tag)
         return tag
 
+    @staticmethod
+    def _validate_tag(tag: List[str]) -> None:
```

Comment:
If you want to be fancy
```suggestion
   TagType = List[str]
    def _validate_tag(tag: TagType ) -> None:
```
and use TagType everywhere...